layers/shader_validation.cpp

```diff
@@ -2366,6 +2366,30 @@ bool CoreChecks::ValidateTransformFeedback(SHADER_MODULE_STATE const *src) const
     return skip;
 }
 
+bool CoreChecks::ValidateWorkgroupInitialization(SHADER_MODULE_STATE const *src, spirv_inst_iter &insn) const {
+    bool skip = false;
+
+    const uint32_t opcode = insn.opcode();
+    if (opcode == spv::OpVariable) {
+        uint32_t storage_class = insn.word(3);
+        if (storage_class == spv::StorageClassWorkgroup) {
+            if (insn.len() > 4 &&
+                !enabled_features.zero_initialize_work_group_memory_features.shaderZeroInitializeWorkgroupMemory) {
+                const char *vuid = IsExtEnabled(device_extensions.vk_khr_zero_initialize_workgroup_memory)
+                                       ? "VUID-RuntimeSpirv-shaderZeroInitializeWorkgroupMemory-06372"
+                                       : "VUID-RuntimeSpirv-OpVariable-06373";
+                skip |= LogError(
+                    device, vuid,
+                    "vkCreateShaderModule(): "
+                    "VkPhysicalDeviceZeroInitializeWorkgroupMemoryFeaturesKHR::shaderZeroInitializeWorkgroupMemory is not enabled, "
+                    "but shader contains an OpVariable with Workgroup Storage Class with an Initializer operand.");
```

Comment:
nit: Make use of new log statement

```suggestion
                    "but shader contains an OpVariable with Workgroup Storage Class with an Initializer operand.\n%s", src->DescribeInstruction(insn).c_str());
```

Reply:
I've been away for too long! Thanks!

Reply:
It's pretty new @ziga-lunarg. I'd probably also forget :)